sdks/fabric/interoperation-node-sdk/src/AssetManager.ts

```diff
@@ -666,11 +703,65 @@ const StartHTLCAssetLockListener = (
-    lockCallback: (c: Contract, d: string, t: string, i: string, r: string, l: string, v: string) => any,
+    lockCallback: (c: Contract, d: string, t: string, i: string, r: string, l: string, v: Hash, timeout: number) => any,
 ): void => {
     StartHTLCEventListener(contract, 'LockAsset', contractId, assetType, assetId, -1, recipientECert, lockerECert, lockCallback);
 }
 
+//TODO: Timed counterpart for timed AssetLockListener for reversion
+const assetLockCallbackTimed = (
+    contract: Contract, 
+    contractID: string, 
+    assetType: string, 
+    assetID: string, 
+    recipientECert: string, 
+    lockerECert: string, 
+    hash: Hash,
+    expiryTime: number
+): void => {
+    //Compare expiryTimeSec with currentTimeSec, which is number of milliseconds since epoch (L174)
+    const currTimeSecs = Math.floor(Date.now()/1000);
+
+    const reclaimCallback = async function(contract: Contract, assetType: string, assetID: string, recipientECert: string, contractID: string) {
+        //TODO: Call isAssetLockedInHTLC -> Hasn't been claimed yet. If true, do nothing. If false, either do the following cases:
+        const [islocked, isLockedQueryError] = await helpers.handlePromise(isAssetLockedInHTLCqueryUsingContractId(contract,contractID));
+        
+        if(islocked == false) {
+            //CASE #1: Check GetHTLCHashPreImageByContractId(contractId)
+            const [result, evaluateError] = await helpers.handlePromise(
+                contract.evaluateTransaction("GetHTLCHashPreImageByContractId", contractID),
+            );
+            
+            //CASE #2: If the function above returns ANY error, call reclaimAssetInHTLC
+            if (evaluateError) {
+                //throw new Error(`GetHTLCHashPreImageByContractId evaluateTransaction Error: ${evaluateError}`);
+                //If it succeeds, no change
+                const [reclaimResult, isReclaimableQueryError] = await helpers.handlePromise(reclaimAssetInHTLCusingContractId(contract, contractID));
+                if(isReclaimableQueryError) {
+                    let i = 0;
+                    do {
+                        reclaimAssetInHTLC(contract, assetType, assetID, recipientECert);
+                        i++;
+                    } while(i < 3);                }
+
+                //Put retry logic in event of failure. Retry 8x and then give up if unsuccessful (temp solution for now; requires CORE changes)
+                //If it fails, retry for i (arbitrary) more times until you quit
+            }
+        }
+    }
+
+    //TODO: (HAVE TIME), Call setTimeout with reclaimCallback
+    if(1000*(expiryTime - currTimeSecs) > 0) {
+        //claimAssetInHTLC(contract, assetType, assetID, recipientECert, hash);
+        setTimeout(reclaimCallback, expiryTime - currTimeSecs, contract, assetType, assetID, recipientECert, hash);
+    }
+
+    //TODO: (NO TIME), Call the reclaim callback immediately
+    else if(1000*(expiryTime - currTimeSecs) <= 0) {
```

Comment:
Spaces after if, before and after the operator *.

Reply:
You don't need the multiplication with 1000 for this comparison. In fact, for readability, change this to `if (expiryTime > currTimeSecs) {`.